### PR TITLE
Aoc2022 day12 solutions

### PR DESCRIPTION
**Pull request by bradcray**
My serial and parallel versions to Advent of Code 2022, day 12, where some notable ones are:

* day12-par.chpl is my favorite parallel version (though it's not very cleaned up)
* day12-par-multiversion.chpl is a version that compares a few different idioms, one of which didn't work until Vass's recent bug fix in #21789 
  * version 0: begin + with statement as a workaround for version 1
  * version 1: begin-only which exhibited a buggy behavior until #21789 was meged
  * version 2: coforall, which seems to use many fewer tasks than either of the above
  * version 3: a serial for version as a sanity check

### CODE REVIEW

File: test/studies/adventOfCode/2022/day12/bradc/day12-par.chpl [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  } while (!Steps[pos].compareExchange(prevDist, steps));
+  if pos == endLoc {
+    return;
+  }
+
+  for off in nextSteps() {
+    const nextPos = pos + off;
+    if Grid.contains(nextPos) {
+      const nextElev = Elevation[nextPos];
+      if nextElev <= elevation || nextElev == elevation+1 {
```
Comment (DanilaFe) [inline test/studies/adventOfCode/2022/day12/bradc/day12-par.chpl:48]: Isn't this `nextEleve <= elevation + 1`?

File: test/studies/adventOfCode/2022/day11/bradc/day11-par.chpl [hunk 2]
```diff
@@ -0,0 +1,115 @@
+use IO, List, Set, Barriers;
```
Comment (cassella) [inline test/studies/adventOfCode/2022/day11/bradc/day11-par.chpl:1]: Should the day11 programs be in this PR too?

Comment (bradcray) [reply test/studies/adventOfCode/2022/day11/bradc/day11-par.chpl:1]: No, that was an operator error that I think was caused by branching from my day11 branch, and should resolve itself once I start merging these in order...

**Comment by vasslitvinov** [comment]
@bradcray I suggest merging it to make sure it works as expected.

**Comment by bradcray** [comment]
Will do.  I haven't failed to merge it due to the bug, just due to lack of time.  I'll try to do this this week.

**Pull request merged**